### ISSUE TITLE
Database session store is too small

### ISSUE THREAD

**Issue by ofriedrich** [simplesamlphp/simplesamlphp#136]
If you configure simplesamlphp to use a database to store the session data, it uses a table like this:

``` php
        $query = 'CREATE TABLE ' . $this->prefix . '_kvstore (_type VARCHAR(30) NOT NULL, _key VARCHAR(50) NOT NULL, _value TEXT NOT NULL, _expire TIMESTAMP, PRIMARY KEY (_key, _type))';
```

The value-column is a text, which means 2^16 bytes max. If you have many attributes (e. g. from LDAP), the value-column is too small, to store the serialized session data. There will be no error, when the data is serialized and stored, but when the data is unserialized after reading.

**Comment by jaimeperez** [comment]
Hi @ofriedrich 

Thanks for telling us about this problem. I'm marking this as an enhancement as this will only hit users in certain cases, basically when they have a lot of attributes or huge ones (like pictures).

I guess using a _LONGTEXT_ type instead should be more than enough (I really hope nobody ends up with a session bigger than 4GB :smile:). The problem I see is backwards-compatibility with all existing databases, so if we change this we should provide a conversion script at least.

Would you be willing to provide a pull request for this?

**Comment by ofriedrich** [comment]
Hi @jaimeperez 

we have LDAP-Users with a huge number of values. Our system is growing and every time we install a new SP - starting from the original source of course - we run into this error.

I see no problem in backwards-compatibility. I think, there is no need, to make an update-script for existing databases, because most users are happy with their existing structure. If you add a message to the readme, it may help users with same problem.

Excuse me, but my English is not so good. What do you mean with "Would you be willing to provide a pull request for this?"

**Comment by jaimeperez** [comment]
Hi @ofriedrich!

Sorry for the delay. What I meant is if you could do the changes necessary yourself, inside your own fork of this repository, and then submit those changes as a pull request that we can merge automatically. In github, a pull request is the equivalent to sending a patch file and asking it to be incorporated into the code base.

Since it is quite difficult for us (if not impossible) to test every single use case, scenario or feature, pull requests are the most effective way to solve issues or add new features without the need to spend hours setting up a test scenario, if those interested (i.e.: you in this case) can already verify that the fix works.

**Comment by tvdijen** [comment]
LONGTEXT will be used by default when available since 1.15

**Comment by ofriedrich** [comment]
@tvdijen Thanks for update.
